lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,510 @@
+  delete testGranule.endingDateTime;
+  delete testGranule.processingStartDateTime;
+  delete testGranule.processingEndDateTime;
+  delete testGranule.lastUpdateDateTime;
+  delete testGranule.timeToArchive;
+  delete testGranule.productionDateTime;
+  delete testGranule.timestamp;
+  delete testGranule.provider;
+  delete testGranule.queryFields;
+  delete testGranule.version;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const record = await granulePgModel.get(knex, {
+    granule_id: testGranule.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omit(record, ['cumulus_id']),
+    {
+      granule_id: testGranule.granuleId,
+      status: testGranule.status,
+      collection_cumulus_id: collectionCumulusId,
+      published: testGranule.published,
+      duration: null,
+      time_to_archive: null,
+      time_to_process: null,
+      product_volume: null,
+      error: null,
+      cmr_link: null,
+      execution_cumulus_id: executionCumulusId,
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: null,
+      ending_date_time: null,
+      last_update_date_time: null,
+      processing_end_date_time: null,
+      processing_start_date_time: null,
+      production_date_time: null,
+      timestamp: null,
+      created_at: new Date(testGranule.createdAt),
+      updated_at: new Date(testGranule.updatedAt),
+    }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateFileRecord handles nullable fields on source file data', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+
+  delete testFile.bucket;
+  delete testFile.checksum;
+  delete testFile.checksumType;
+  delete testFile.fileName;
+  delete testFile.key;
+  delete testFile.path;
+  delete testFile.size;
+  delete testFile.source;
+
+  const granuleCumulusId = await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, granuleCumulusId, knex);
+
+  const record = await filePgModel.get(knex, { bucket: null, key: null });
+
+  t.deepEqual(
+    omit(record, fileOmitList),
+    {
+      bucket: null,
+      checksum_value: null,
+      checksum_type: null,
+      file_size: null,
+      file_name: null,
+      key: null,
+      source: null,
+      path: null,
+    }
+  );
+});
+
+test.serial('migrateGranulesAndFiles skips already migrated granule record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await Promise.all(testGranule.files.map((file) => s3Utils.s3PutObject({
```

Comment:
Do we have to stage files in S3 for the migration to work?

Reply:
For some reason that escapes me now I that we needed to do that but it doesn't seem to be the case anymore... 🤔